specs/Spec.Hash.PadFinish.fst

```diff
@@ -7,8 +7,19 @@ open Lib.ByteSequence
 open Spec.Hash.Lemmas0
 open Spec.Hash.Definitions
 
-(** This module contains specifications shared across all the Merkle-Damgård
-    constructions. *)
+(** This module contains two things.
```

Comment:
should we move finish to Spec.Hash.Incremental and pad to Spec.Hash.MD? (this module is weirdly designed)

Reply:
Yes, this has been bothering me for a while as well. The proposed plan looks reasonable.
Some other related cleanups:
* Spec.Hash.Lemmas should go in specs/lemmas
* Since Spec.Hash.Lemmas now only contains lemmas about update_multi, it does not need to open Spec.Hash.PadFinish, and the extra Spec.Hash.Lemmas0 does not seem needed anymore.

Reply:
👍